File: wolt/src/main/java/io/helidon/kotlin/service/wolt/DeliveryMapper.java
package io.helidon.kotlin.service.wolt;

import io.helidon.dbclient.DbColumn;
import io.helidon.dbclient.DbMapper;
import io.helidon.dbclient.DbRow;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class DeliveryMapper implements DbMapper<Delivery> {

    @Override
    public Delivery read(DbRow row) {
        DbColumn id = row.column("id");
        DbColumn food = row.column("food");
        DbColumn address = row.column("address");
        DbColumn status = row.column("status");
        return new Delivery(id.as(String.class), food.as(String.class), address.as(String.class), status.as(DeliveryStatus.class));
    }

    @Override
    public Map<String, Object> toNamedParameters(Delivery value) {
        Map<String, Object> map = new HashMap<>(1);
        map.put("id", value.getId());
        map.put("food", value.getFood());
        map.put("address", value.getAddress());
        map.put("status", value.getDeliveryStatus().toString());
        return map;
    }

    @Override
    public List<Object> toIndexedParameters(Delivery value) {
        List<Object> list = new ArrayList<>(4);
        list.add(value.getId());
        list.add(value.getFood());
        list.add(value.getAddress());
        list.add(value.getDeliveryStatus().toString());
        return list;
    }
}

File: wolt/src/main/java/io/helidon/kotlin/service/wolt/DeliveryMapper.kt
package io.helidon.kotlin.service.wolt

import io.helidon.dbclient.DbMapper
import io.helidon.dbclient.DbRow
import support.to
import java.util.ArrayList
import java.util.HashMap


class DeliveryMapper : DbMapper<Delivery> {

    override fun read(row: DbRow): Delivery {
        val id = row.column("id").to<String>()
        val food = row.column("food").to<String>()
        val address = row.column("address").to<String>()
        val status = row.column("status").to<DeliveryStatus>()
        return Delivery(
            id, food, address, status
        )
    }

    override fun toNamedParameters(value: Delivery): Map<String, Any> {
        val map: MutableMap<String, Any> = HashMap(1)
        map["id"] = value.id
        map["food"] = value.food
        map["address"] = value.address
        map["status"] = value.deliveryStatus.toString()
        return map
    }

    override fun toIndexedParameters(value: Delivery): List<Any> {
        val list: MutableList<Any> = ArrayList(4)
        list.add(value.id)
        list.add(value.food)
        list.add(value.address)
        list.add(value.deliveryStatus.toString())
        return list
    }
}